### ISSUE TITLE
Empty groups in addressbook silently not exported

### ISSUE THREAD

**Issue by respiranto** [roundcube/roundcubemail#9590]
### Prerequisites

- [X] I have [searched](https://github.com/roundcube/roundcubemail/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I can recreate the issue with all plugins disabled

### Describe the issue

When exporting an addressbook via the Roundcube web UI ("Export all"), any group without members is silently ignored.

What I would expect is either that empty groups are exported, or a warning message indicating that the respective groups without members could not be exported and why.

In particular, I would expect that the result of "Export all" (if no error is indicated) can be used with "Import" to recreate the original addressbook.

Given that Roundcube exports to the vCard 3.0 format (c.f. [RFC 2426](https://datatracker.ietf.org/doc/html/rfc2426)), and represents groups as values of `CATEGORIES` for each group member, it seems impossible to actually export a group without members.

Hence, I suggest a warning message specifying that certain to be named groups could not be exported because not having any members if that is the case.

---
Originally reported at the Debian BTS as [Bug 1078777](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1078777), where I was asked to report the bug here instead.

### What browser(s) are you seeing the problem on?

Firefox

### What version of PHP are you using?

v8.2

### What version of Roundcube are you using?

master / 58721e3037671888f82d75fd8e91a8622405af45

**Comment by alecpl** [comment]
vCard 4.0 solves the problem, but of course it's not a simple solution. A warning makes sense.

**Comment by respiranto** [comment]
Is vCard 4.0 support planned?

Relatedly, might I suggest using an external vCard library (e.g., [sabre/vobject](https://sabre.io/vobject/)) to help address this and the other vCard-related bugs?

**Comment by pabzm** [comment]
> Is vCard 4.0 support planned?

Not really. But I think we wouldn't oppose it if someone would implement it.

So (unless you want to tackle the job) I'd agree that a simple warning would be the best option. Would you like to implement that?
